### PR TITLE
Mention `fields` function in docs

### PR DESCRIPTION
**Pull request by rikhuijzer**
While doing typst scripting, I often wonder: _what fields can I überhaupt access here?_ 

This PR suggests to mention the function on the `scripting` documentation page (<https://typst.app/docs/reference/scripting/>).

(By the way, I'm having a blast writing my PhD thesis in typst. I used to always carry some exotic LaTeX template around to get the document to print the way my field required, but with typst I have solved my styling in about 40 lines and a few hours. No need for hours of tuning nor thousands of lines of code. Amazing work by you on figuring out the right abstractions!)

### CODE REVIEW

**Comment by laurmaedje** [comment]
could you rephrase it a bit to clarify that `.fields()` is only available on `content` and not on every type?

**Comment by laurmaedje** [comment]
> By the way, I'm having a blast writing my PhD thesis in typst. I used to always carry some exotic LaTeX template around to get the document to print the way my field required, but with typst I have solved my styling in about 40 lines and a few hours. No need for hours of tuning nor thousands of lines of code. Amazing work by you on figuring out the right abstractions!

Thanks, it means a lot :)

File: docs/reference/scripting.md [hunk 1]
```diff
@@ -231,8 +231,10 @@ The body of a loop can be a code or content block:
 - `{while condition [..]}`
 
 ## Fields
-You can use _dot notation_ to access fields on a value. The value in question
-can be either:
+You can use _dot notation_ to access fields on a value. For values of type
+`content`, you can also use the `fields` function to list the fields.
```
Comment (laurmaedje) [inline docs/reference/scripting.md:235]:
```suggestion
[`content`], you can also use the [`fields`]($content.fields) function to list
the fields.
```

**Comment by laurmaedje** [comment]
Thanks!

**Comment by rikhuijzer** [comment]
Thanks for the review!
